src/include/miopen/groupnorm/problem_description.hpp

```diff
@@ -0,0 +1,162 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <miopen/problem_description_base.hpp>
+#include <miopen/activ.hpp>
+#include <miopen/tensor.hpp>
+
+#include <cassert>
+#include <string>
+
+namespace miopen {
+
+struct NetworkConfig;
+
+namespace groupnorm {
+
+struct ProblemDescription : ProblemDescriptionBase
+{
+    ProblemDescription(miopenNormMode_t mode_,
+                       const TensorDescriptor& xDesc_,
+                       const TensorDescriptor& weightDesc_,
+                       const TensorDescriptor& biasDesc_,
+                       const TensorDescriptor& yDesc_,
+                       const TensorDescriptor& meanDesc_,
+                       const TensorDescriptor& rstdDesc_,
+                       uint64_t num_groups_,
+                       float epsilon_)
+        : mode(mode_),
+          xDesc(xDesc_),
+          weightDesc(weightDesc_),
+          biasDesc(biasDesc_),
+          yDesc(yDesc_),
+          meanDesc(meanDesc_),
+          rstdDesc(rstdDesc_),
+          num_groups(num_groups_),
+          epsilon(epsilon_)
+    {
+        if(xDesc.GetType() != yDesc.GetType())
+        {
+            MIOPEN_THROW(miopenStatusBadParm, "GroupNormForward: Tensor types do not match.");
```

Comment:
Is it really invalid case?

Reply:
Yes because I assumed their types to be same in the `GroupNormFwdContiguous` kernel. 
```
extern "C" __global__ void GroupNormFwdContiguous(const FLOAT* __restrict__ x,
                                                  FLOAT* __restrict__ y,
                                                  const FLOAT* __restrict__ weight,
                                                  const FLOAT* __restrict__ bias,
                                                  FLOAT_ACCUM* __restrict__ mean,
                                                  FLOAT_ACCUM* __restrict__ rstd,
                                                  float eps,
                                                  uint64_t num_groups,
                                                  uint64_t num_channels,
                                                  uint64_t numel_per_channel,
                                                  bool mode)
```

Reply:
@kyeonghwanryu 
> Yes because I assumed their types to be same in the GroupNormFwdContiguous kernel.

But if kernel does not this, then it doesn't mean that the problem is invalid. It only means that this specific solver is unable to find a solution for this problem (and in this case it's IsApplicable should return false).

>> Is it really invalid case?

Let me rephrase the question: If the types of x and y are not the same, does that mean the problem is invalid? I do not think so, but I believe you know better the semantics of the Group Norm primitive.

Reply:
Fixed as per your comment and added missing checks for weight and bias.